internal/mode/static/state/graph/grpcroute_test.go

```diff
@@ -310,6 +328,36 @@ func TestBuildGRPCRoute(t *testing.T) {
 			},
 			name: "valid rule with empty match",
 		},
+		{
+			validator: createAllValidValidator(),
+			gr:        grValidFilter,
+			expected: &L7Route{
+				RouteType: RouteTypeGRPC,
+				Source:    grValidFilter,
+				ParentRefs: []ParentRef{
+					{
+						Idx:         0,
+						Gateway:     gatewayNsName,
+						SectionName: grValidFilter.Spec.ParentRefs[0].SectionName,
+					},
+				},
+				Valid:      true,
+				Attachable: true,
+				Spec: L7RouteSpec{
+					Hostnames: grValidFilter.Spec.Hostnames,
+					Rules: []RouteRule{
+						{
+							ValidMatches:     true,
+							ValidFilters:     true,
+							Matches:          convertGRPCMatches(grValidFilter.Spec.Rules[0].Matches),
+							RouteBackendRefs: []RouteBackendRef{},
+							Filters:          convertGRPCFilters(grValidFilter.Spec.Rules[0].Filters),
```

Comment:
because we use convertGRPCFillters here, we don't really test that function.

Would it make sense to have a separate unit test for convertGRPCFilters?
And probably convertGRPCMatches?

Alternatively, instead of `Filters:          convertGRPCFilters`..., we can just define structs there

Reply:
Added a separate unit-test for convertGRPCMatches as it's more complex, and defined a struct for `Filters` instead of calling the convert function